### PR TITLE
feat(query): add aggregation any metric

### PR DESCRIPTION
**Pull request by Ahoo-Wang**
## Goal

Allow snapshot aggregation results to display a denormalized value, such as `productName`, without adding it to the grouping key. Completion means `terms(productId) + any(productName)` returns one row per product ID, chooses a non-null scalar when available, and behaves consistently across the MongoDB and Elasticsearch query services.

## Changes

- Add public `AggregationMetric.Any`, JSON subtype `ANY`, and Kotlin DSL `any(field, alias)`.
- Validate `ANY` fields through the existing `AGGREGATE_TERMS` capability and reject known multi-valued fields through the current schema validation modes.
- Compile MongoDB `ANY` to native `$max` and Elasticsearch `ANY` to `terms(size = 1)` without scripts, runtime fields, `_source`, or `top_hits`.
- Bound Elasticsearch composite page size by the number of `ANY` sub-buckets to stay within the existing bucket budget.
- Preserve metric sorting and HTTP expensive-query guard behavior.
- Expose the new metric union through OpenAPI and make explicitly `READ_WRITE` Kotlin getter-only properties writable in generated schemas.
- Add shared real-backend tests for inconsistent names, mixed null/non-null values, all-null values, and empty summaries.
- Update English and Chinese query documentation, OpenAPI snapshots, and schema golden files.

## Verification

- `./gradlew build --stacktrace` — passed, 334 actionable tasks.
- `./gradlew :wow-mongo:integrationTest :wow-elasticsearch:integrationTest --rerun-tasks --stacktrace` — passed, 43/43 tasks executed.
- `./gradlew detekt :wow-api:check :wow-query:check :wow-mongo:check :wow-elasticsearch:check :wow-webflux:check :wow-openapi:check :wow-schema:check --stacktrace` — passed.
- `cd documentation && pnpm docs:build` — passed.
- Final whole-feature review and scoped fix re-reviews found no remaining Critical, Important, or Minor issues.

## Compatibility and risks

- [ ] This change preserves public API and generated contract compatibility.
- [x] Tests cover the changed behavior or the verification alternative is explained.
- [x] Documentation is updated when user-visible behavior changes.
- [x] Comments and API documentation explain non-obvious constraints and remain accurate after this change.
- [x] No credentials, generated build output, or local environment files are included.

Compatibility is intentionally not preserved: adding a subtype to the public sealed `AggregationMetric` hierarchy requires downstream exhaustive `when` expressions to handle `Any`, and generated OpenAPI clients see the new metric union/base reference.

`ANY` is intentionally non-deterministic across executions and backends. MongoDB uses `$max`; Elasticsearch selects one terms bucket. Callers that require a stable display name must repair the denormalized data or use an explicit business query.

Elasticsearch reduces per-page composite rows as the number of `ANY` metrics grows. This prevents bucket-limit failures at the cost of additional PIT pagination requests for wide aggregation queries.

No storage migration, deployment, release, or production rollout is included. Rollback is a normal revert of this PR.

### CODE REVIEW

**Comment by Ahoo-Wang** [comment]
@codex

File: wow-elasticsearch/src/main/kotlin/me/ahoo/wow/elasticsearch/query/snapshot/ElasticsearchAggregationPager.kt [hunk 1]
```diff
@@ -221,8 +236,23 @@ internal class ElasticsearchAggregationPager(
...
-        val function = function ?: return docCount
+    ): Any? = when (this) {
+        is ElasticsearchAggregationMetric.Count -> docCount
+        is ElasticsearchAggregationMetric.Any -> aggregations.getValue(alias).anyValue(alias)
+        is ElasticsearchAggregationMetric.Numeric -> numericValue(aggregations)
+    }
+
+    private fun Aggregate.anyValue(alias: String): Any? = when {
+        isSterms -> sterms().buckets().array().firstOrNull()?.key()?.nativeValue()
+        isLterms -> lterms().buckets().array().firstOrNull()?.key()
```
[the remark this answers is not in the document]
Comment (Ahoo-Wang) [reply wow-elasticsearch/src/main/kotlin/me/ahoo/wow/elasticsearch/query/snapshot/ElasticsearchAggregationPager.kt]: Fixed in `08d1dced9`. The `lterms` decoder now parses `key_as_string` with `toBooleanStrictOrNull()` and falls back to the numeric key for non-Boolean long terms. The regression fixture now mirrors Elasticsearch Boolean terms (`key` 0/1 plus `key_as_string` false/true) and asserts both values. Verified with `:wow-elasticsearch:check` and the repository coverage task.

File: wow-api/src/main/kotlin/me/ahoo/wow/api/query/AggregationQuery.kt [hunk 2]
```diff
@@ -212,6 +222,15 @@ sealed interface AggregationMetric {
             requireAggregationAlias(alias)
         }
     }
+
+    data class Any(
+        val field: LogicalField,
+        override val alias: String,
+    ) : AggregationMetric {
```
[the remark this answers is not in the document]
Comment (Ahoo-Wang) [reply wow-api/src/main/kotlin/me/ahoo/wow/api/query/AggregationQuery.kt:229]: Verified. This change intentionally crosses a declared source-compatibility boundary: the accepted scope explicitly does not preserve compatibility for the public aggregation metric hierarchy. `ANY` therefore remains an explicit request-model subtype and discriminator instead of being hidden behind an existing field or an implicit switch.
